### ISSUE TITLE
RecursionError when checking code with `pandas.merge`

### ISSUE THREAD

**Issue by Lee-W** [PyCQA/pylint#4024]
### Steps to reproduce
1. `pip install pylint==2.6.0 pandas==1.1.5 git+https://github.com/PyCQA/astroid.git@4b7566b0c8365613198af493d4115d32d7d4c66e`
2. Run `pylint` to check the following code

```python
import pandas
pandas.merge(None, None)
```

This bug relates to #3318 and https://github.com/PyCQA/astroid/pull/804 (This is why I use that astroid version)

### Current behavior
```
RecursionError: maximum recursion depth exceeded while calling a Python object
```

### Expected behavior
Check python program with running into error

### pylint --version output

```
pylint 2.6.0
astroid 2.5.0
Python 3.8.3 (default, Jul 29 2020, 12:45:34)
[Clang 11.0.0 (clang-1100.0.33.8)]
```

**Comment by dbaty** [comment]
Unless I missed something, this is a duplicate of #3977. I can reproduce the error with pandas 1.1.5 and the master branches of both pylint and astroid. (Note: the error occurs only with pandas **1.1.5**. With the latest version of pandas (1.2.0), there is no error.)

**Comment by hippo91** [comment]
@Lee-W thanks for the report and @dbaty thanks for your investigation!
I close this issue.
